MetaMask/metamask-mobile
# feat: MultichainBalanceController

## **Description**
Builds off of the work done in [this PR](https://github.com/MetaMask/core/pull/4965).

- This PR adds the ability to see your non EVM token balance and the fiat token rates (Solana and Bitcoin accounts)
- This is achieved by using the RatesController and the MultichainBalanceController
- This PR also extract the balance logic from the PortfolioBalance PR into the useMultichainBalances hook
- creates the selectors for multichain non evm

This PR DOES NOT...
1. show the correct native token balance in the account picker. This will be done in a [following PR](https://github.com/MetaMask/accounts-planning/issues/697) that builds off of this work.
2. enable token details for these non evm assets. (for now it will simply show no data)

## **Related issues**

Fixes: https://github.com/MetaMask/accounts-planning/issues/696

## **Manual testing steps**

1. checkout this branch
3. open .js.env
4. set export METAMASK_BUILD_TYPE="flask" // flask
5. source .js.env
6. yarn setup
7. yarn start:ios
8. create/import a wallet
9. open the account menu actions
10. you should see that you can add a BTC and BTC Testnet account
11. click Add a new Bitcoin account (Beta)
12. a popup with a suggested name should appear click confirm
13. the newly selected account should be your Bitcoin account and the network picker should show the Bitcoin logo the network should not be clickable
14. Optionally, create a testnet account and fund it with this faucet: https://bitcoinfaucet.uo1.net/
15. If you have a balance, you should see the fiat balance rendered on the wallet
16. if not you have a balance, then you should see a 0 fiat balance.
17. If you change your selected currency (settings/general) and then return to the main wallet view, the balance should update based on the new fiat currency.
18. The token list should also list your account balance in both fiat and and the native token balance.
19. the token displayed should be the BTC logo and not have any network badge above it (like the eth account does)
20. now add a bitcoin TestNet account.
21. the portfolio balance should show the native token balance not the fiat balance.
22. repeat the above steps but this time add a new Solana account. If your Solana account has a balance, it should be rendered on the home page in fiat. If not you should see a 0 balance.
23. we do not currently support Solana testnet accounts.


## **Screenshots/Recordings**



### **Before**



### **After**

https://github.com/user-attachments/assets/12d4bfeb-608f-4744-9c0a-459163eb0dba



<img src="https://github.com/user-attachments/assets/f8428b91-fb45-433d-921e-3824845a71ef" height="400" height="600"/>



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
